private/serialization.nim

```diff
@@ -62,12 +62,40 @@ proc representObject*(value: string, ts: TagStyle,
   ## represents a string as YAML scalar
   c.put(scalarEvent(value, tag, yAnchorNone))
 
+proc parseHex[T: int8|int16|int32|int64|uint8|uint16|uint32|uint64](s: string): T =
+  result = 0
+  var i = 2
+  while true:
+    case s[i]
+    of '_': discard
+    of '0'..'9': result = result shl 4 or (T(ord(s[i]) - ord('0')) and 0xFF)
+    of 'a'..'f': result = result shl 4 or (T(ord(s[i]) - ord('a') + 10) and 0xFF)
+    of 'A'..'F': result = result shl 4 or (T(ord(s[i]) - ord('A') + 10) and 0xFF)
+    of '\0': break
+    else: raise newException(ValueError, "Invalid character in hex: " & s[i])
+    inc(i)
+
+proc parseOctal[T: int8|int16|int32|int64|uint8|uint16|uint32|uint64](s: string): T =
+  var i = 2
+  while true:
+    case s[i]
+    of '_': discard
+    of '0'..'8': result = result * 8 + T((ord(s[i]) - ord('0')))
+    of '\0': break
+    else: raise newException(ValueError, "Invalid character in hex: " & s[i])
+    inc(i)
+
 proc constructObject*[T: int8|int16|int32|int64](
     s: var YamlStream, c: ConstructionContext, result: var T)
     {.raises: [YamlConstructionError, YamlStreamError].} =
   ## constructs an integer value from a YAML scalar
   constructScalarItem(s, item, T):
-    result = T(parseBiggestInt(item.scalarContent))
+    if item.scalarContent[0] == '0' and (item.scalarContent[1] == 'x' or item.scalarContent[1] == 'X'):
```

Comment:
`item.scalarContent[1] in {'x', 'X'}` would be nice here. even nicer if `{'x', 'X'}` was a const, like, `hexIndicator` or something.